### DISCUSSION TITLE
On -c and -i flags

### DISCUSSION THREAD

**Discussion by yohannd1** [janet-lang/janet#1260] [General]
Hello! I'm new to Janet and I've noticed the `-c` and `-i` flags on the `janet` executable. What are they for, exactly? I can't find much about it on the documentation and my guess (that `-c` compiles and `-i` executes bytecode) doesn't seem to apply because:
```
$ echo '(print "Hello, World!")' > a.janet

$ janet -c a.janet out
Hello, World!

$ janet -i out
```

**Comment by CosmicToast** [comment]
Try like this:
```bash
$ echo '(defn main [_] (print "Hello, World!"))' > a.janet
$ janet -c a.janet out
$ janet -i out
```

Same semantics as compiling to an executable apply.

**Reply by yohannd1** [reply]
Oh. Thanks!

**Comment by sogaiu** [comment]
Also of interest might be the [Compilation and Imagination chapter](https://janet.guide/compilation-and-imagination/) of @ianthehenry's [Janet for Mortals](https://janet.guide).

**Reply by yohannd1** [reply]
Oh, I hadn't heard about that guide. Thanks!

**Reply by primo-ppcg** [reply]
> Janet for Mortals

That's the most entertaining language guide I've ever read 👍
